frontend/src/pages/Overview.js

```diff
@@ -31,12 +33,54 @@ const FlexColumn = styled.div`
 
 const Overview = () => {
     const [t] = useTranslation(['parentData']);
+    const { seriesId } = useParams();
+
+    const [
+        {
+            parentData: { seriesData, buildData },
+            branchesState,
+        },
+        dispatch,
+    ] = useStateValue();
 
     const pathname = useLocation().pathname;
     const buildUrl = pathname.includes('build');
 
     const status = buildUrl ? 'build' : 'series';
 
+    useEffect(() => {
+        if (branchesState) {
+            const branch = branchesState.series?.find(
+                ({ id: serie_id }) => serie_id === parseInt(seriesId, 10)
+            );
+            const fetchData = async () => {
+                dispatch({ type: 'setSeriesData', seriesData: branch });
+                dispatch({
+                    type: 'setSelectedBranch',
+                    name: branch.name || ' ',
+                    id: seriesId,
+                    team: branch.team || ' ',
+                });
+                try {
+                    const { last_build } = branch;
+                    const buildUrl = `/data/series/${seriesId}/builds/${last_build}/info?`;
+                    const res = await fetch(buildUrl);
+                    const json = await res.json();
+                    const buildData = json.build;
+
+                    dispatch({ type: 'setBuildData', buildData });
+                } catch (error) {
+                    dispatch({ type: 'setErrorState', errorState: error });
+                }
+            };
+
+            fetchData();
+        }
+        return () => {
+            dispatch({ type: 'flushParentData' });
+        };
+    }, [dispatch, seriesId, branchesState]);
```

Comment:
same question here. Also i see same code as above. Maybe we should move it to some file where we call it ?

Reply:
Yea, not sure where that file would be located though as these useEffects are called on every state after the selection of series. 
I have now changed the location of this useEffect to the series file of overview so these values are never loaded twice.